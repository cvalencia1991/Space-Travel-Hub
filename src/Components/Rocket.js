/* eslint-disable react/prop-types */
/* eslint-disable camelcase */
import { React } from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { CardImg } from 'react-bootstrap';

const Rocket = (props) => {
  const {
    id, rocket_name, description, flickr_images,
  } = props;

  return (
    <Card style={{ width: '100%' }}>
      <Row>
        <Col>
          <CardImg src={flickr_images[0]} />
        </Col>
        <Col>
          <Card.Body>
            <Card.Title>Card Title</Card.Title>
            <Card.Text>
              Some quick example text to build on the card title and make up the
              bulk of the card&aposs content.
            </Card.Text>
            <Button variant="primary">Go somewhere</Button>
          </Card.Body>
        </Col>
      </Row>
    </Card>
  );
};

export default Rocket;