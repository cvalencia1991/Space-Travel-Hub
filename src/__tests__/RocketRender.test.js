import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rocket from '../Components/Rocket';
import store from '../Redux/configureStore';

it('renders correctly', () => {
  const rocketComponent = renderer
    .create(
      <Provider store={store}>
        <Rocket
          key="1"
          id="1"
          rocketName="Test Rocket"
          description="Test description"
          flickrImages="https://imgur.com/DaCfMsj.jpg"
          reserved="false"
        />

      </Provider>,
    )
    .toJSON();
  expect(rocketComponent).toMatchSnapshot();
});
