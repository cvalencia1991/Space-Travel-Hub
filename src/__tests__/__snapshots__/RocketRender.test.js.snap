// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="border-0 mb-4 card"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <img
        className="card-img"
        src="h"
        style={
          Object {
            "borderRadius": "0",
          }
        }
      />
    </div>
    <div
      className="col"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Test Rocket
        </div>
        <p
          className="card-text"
        >
          <span
            className="badge bg-info"
          >
            Reserved
          </span>
           
          Test description
        </p>
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  </div>
</div>
`;
